### PR TITLE
doc update to make it easy to find how to do re-indexing or delta ingestion

### PR DESCRIPTION
**Pull request by himanshug**
received this feedback from users that it was hard to find so making very explicit.

### CODE REVIEW

File: docs/content/toc.textile [hunk 1]
```diff
@@ -16,7 +16,7 @@ h2. Data Ingestion
 * "Schema Design":../ingestion/schema-design.html
 * "Schema Changes":../ingestion/schema-changes.html
 * "Realtime Ingestion":../ingestion/realtime-ingestion.html
-* "Batch Ingestion":../ingestion/batch-ingestion.html
+* "Batch ingestion or Re-indexing or Delta ingestion":../ingestion/batch-ingestion.html
```
Comment (fjy) [inline docs/content/toc.textile]: I think Re-indexing should be a completely different section in the ingestion docs with these explicit steps listed out

Comment (himanshug) [inline docs/content/toc.textile]: you mean adding another file batch-delta-ingestion.md and link it in "Ingestion" section in the TOC file ? But that new file will only say what is updated in "docs/content/ingestion/batch-ingestion.md
" in this PR. is that enough or you are looking for something more?

Comment (fjy) [inline docs/content/toc.textile]: i think we should add batch-delta-ingestion.md and link it from the TOC. The document can be short but we can move the contents around reingestion out of batch-ingestion or repeat the relevant parts

Comment (himanshug) [inline docs/content/toc.textile]: sure, will update soon.

**Comment by himanshug** [comment]
@fjy changed as per review comments.

File: docs/content/toc.textile [hunk 2]
```diff
@@ -17,6 +17,7 @@ h2. Data Ingestion
 * "Schema Changes":../ingestion/schema-changes.html
 * "Realtime Ingestion":../ingestion/realtime-ingestion.html
 * "Batch Ingestion":../ingestion/batch-ingestion.html
+* "Batch Re-Indexing and Delta-Ingestion":../ingestion/batch-delta-ingestion.html
```
Comment (fjy) [inline docs/content/toc.textile]: How do you feel about calling this 'Updating Existing Data'?

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+layout: doc_page
+---
+
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: How do you feel about giving this section a title 'Updating Existing Data'?

And then we can explain updates in Druid are done by reindexing or rebuilding segments, and by delta ingestion. We can also mention that lookups provide a way to do updates on a small dimension table.

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+This document assumes the reader has understanding of [hadoop batch ingestion](batch-ingestion.html). You can use hadoop batch-ingestion to do re-indexing and delta-ingestion as well.
+
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: I think it is useful to say that updates are best done through batch ingestion, and this section will focus on Hadoop based batch ingestion.

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            "dataSource": "wikipedia",
+            "intervals": ["2014-10-20T00:00:00Z/P2W"]
+        }
+    },
+    {
+        "type" : "static",
+        "paths": "/path/to/more/wikipedia/data/"
+    }
+]
+```
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: I think examples of a full spec will be really helpful

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+|-----|----|-----------|--------|
+|ingestionSpec|Json Object|Specification of druid segments to be loaded. See below.|yes|
+|maxSplitSize|Number|Enables combining multiple segments into single Hadoop InputSplit according to size of segments. Default is none. |no|
+
+Here is what goes inside "ingestionSpec"
+
+|Field|Type|Description|Required|
+|-----|----|-----------|--------|
+|dataSource|String|Druid dataSource name from which you are loading the data.|yes|
+|interval|String|This is deprecated, please use intervals.|no|
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: let's just remove this

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+"ingestionSpec" :
+    {
+        "dataSource": "wikipedia",
+        "intervals": ["2014-10-20T00:00:00Z/P2W"]
+    }
+```
+
+#### `multi`
+
+It is a composing inputSpec to combine two other input specs. It is useful for doing "delta ingestion".
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: why is "delta ingestion" in quotes?

File: docs/content/ingestion/batch-delta-ingestion.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
+---
+layout: doc_page
+---
+
+This document assumes the reader has understanding of [hadoop batch ingestion](batch-ingestion.html). You can use hadoop batch-ingestion to do re-indexing and delta-ingestion as well.
+
+Reindexing means, suppose, you ingest some raw data to a dataSource A and later you want to re-index and create another dataSource B which has a subset of columns or different granularity. Or, you may want to change granularity of data in A itself for some interval.
+
+Delta-ingestion means, suppose, you ingested some raw data to a dataSource A in an interval, later you want to "append" more data to same interval. Note that this is not idempotent operation, we might add some features in future to make it idempotent.
```
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: mention the common case of delayed events

File: docs/content/ingestion/batch-delta-ingestion.md (hunk 3 above)
Comment (fjy) [inline docs/content/ingestion/batch-delta-ingestion.md]: Also I think this is a nice place to mention that updates can be an expensive process in Druid depending on what is done

File: docs/content/toc.textile (hunk 2 above)
Comment (himanshug) [inline docs/content/toc.textile]: changed

**Comment by himanshug** [comment]
@fjy made various changes to address all review comments, pls take another look. I have placed links for users to learn details about ingestion spec etc, this doc focuses on data updation only.

**Comment by fjy** [comment]
:+1:

**Pull request merged**